tests/by-util/test_join.rs

```diff
@@ -350,6 +354,18 @@ fn non_unicode() {
         .arg("non-unicode_2.bin")
         .succeeds()
         .stdout_only_fixture("non-unicode.expected");
+
+    #[cfg(unix)]
+    {
+        let invalid_utf8: u8 = 167;
+        new_ucmd!()
+            .arg("-t")
+            .arg(OsStr::from_bytes(&[invalid_utf8]))
+            .arg("non-unicode_1.bin")
+            .arg("non-unicode_2.bin")
+            .succeeds()
+            .stdout_only_fixture("non-unicode_sep.expected");
+    }
```

Comment:
it would be nice to update the test on non unix to verify that the error is here :)

Reply:
Rebased and pushed with a new test. We have to use Windows for our non-unix, since that's the only other OS that gives any sort of access to the internal `OsString` representation.